### PR TITLE
add integration topology test

### PR DESCRIPTION
**Pull request by yaoliclshlmch**
This request is related to #2936, which was accidentally closed.

### CODE REVIEW

File: heron/common/src/cpp/basics/strutils.cpp [hunk 1]
```diff
@@ -44,3 +44,38 @@ StrUtils::split(
...
+  }
+  return output;
+}
+
+std::vector<char> StrUtils::decode(const std::vector<char>& _input) {
+  std::vector<char> output;
+  int i = 0;
+  while (i < _input.size()) {
+    char chr_1 = _input[i];
+    if (chr_1 <= 57) {
```
Comment (nwangtw) [inline heron/common/src/cpp/basics/strutils.cpp]: Give 57, 48 and 55 const variables to be more readable.

File: heron/common/src/cpp/basics/strutils.cpp [hunk 2]
```diff
@@ -44,3 +44,58 @@ StrUtils::split(
...
+}
+
+// Return decoded value in [0, 15]; return -1 if the character is illegal.
+int StrUtils::decodeHexChar(char c) {
+  const int char_zero = 48;
+  const int char_nine = 57;
+  const int char_A = 65;
+  const int char_F = 70;
+
+  if (c < char_zero) {
```
Comment (nlu90) [inline heron/common/src/cpp/basics/strutils.cpp]: could you simplify it to `if (0 to 9) ... else if (A to F) ... else return -1`?

File: scripts/run_integration_topology_test.sh [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+set -e
+
+# building tar packages
+DIR=`dirname $0`
+source ${DIR}/detect_os_type.sh
+bazel run --config=`platform` -- scripts/packages:heron-install.sh --user
+bazel build --config=`platform` {heron/...,scripts/packages:tarpkgs,integration_test/src/...}
+
+# run the simple http server
+#${HTTP_SERVER} 8080 &
```
Comment (nlu90) [inline scripts/run_integration_topology_test.sh]: any specific reason we keep the HTTP server, scala test,  python test commented out?

Comment (yaoliclshlmch) [reply scripts/run_integration_topology_test.sh]: Http server will be used in instance state validation in stateful processing, and scala and python tests may be added in the future.

**Comment by huijunwu** [comment]
ship after ci pass

**Pull request merged**